app/javascript/src/utils/codemirror/indent.js

```diff
@@ -49,10 +49,12 @@ export function tabIndent({ state, dispatch }, event) {
       //'line.from' and 'from' are equal at start of line, dont reduce indents lower than 0.
       const fromModifier = line.from === from ? 0 : (insert.search(/\S/) - leadingWhitespaceLength - (from === to ? 1: 0))
       const toModifier = insert.length - originalLength
+      const selectionDirection = state.selection.main.head === to
 
       return {
         changes: { from: line.from, to, insert },
-        range: EditorSelection.range(from + fromModifier, to + toModifier)
+        range: selectionDirection ? EditorSelection.range(from + fromModifier, to + toModifier) :
+          EditorSelection.range(to + toModifier, from + fromModifier)
```

Comment:
Maybe we can make it slightly more readable by separating it to a var and use a name that implies a bool (This suggestion won't work via github, sorry)

```suggestion
const reverseSelection = state.selection.main.head === to

let range = EditorSelection.range(to + toModifier, from + fromModifier)
if (reverseSelection) range = EditorSelection.range(from + fromModifier, to + toModifier)

return {
  changes: { from: line.from, to, insert },
  range
```